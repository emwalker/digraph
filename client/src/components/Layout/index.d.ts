import React, { ReactNode } from 'react';
import { Match, Router } from 'found';
import { LayoutQuery$data as Response } from '__generated__/LayoutQuery.graphql';
type AlertsType = Response['alerts'];
type ViewType = Response['view'];
type Props = {
    alerts: AlertsType;
    children?: ReactNode;
    router: Router;
    match: Match;
    view: ViewType;
};
export declare const query: import("react-relay").GraphQLTaggedNode;
export default function Layout({ alerts, children, view, match, router }: Props): React.JSX.Element;
export {};
