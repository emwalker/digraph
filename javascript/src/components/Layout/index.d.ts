import { ReactNode } from 'react';
import { Match, Router } from 'found';
import { LayoutQueryResponse } from '__generated__/LayoutQuery.graphql';
declare type AlertsType = LayoutQueryResponse['alerts'];
declare type ViewType = LayoutQueryResponse['view'];
declare type Props = {
    alerts: AlertsType;
    children?: ReactNode;
    router: Router;
    match: Match;
    view: ViewType;
};
export declare const query: import("react-relay").GraphQLTaggedNode;
declare const Layout: ({ alerts, children, view, match, router }: Props) => JSX.Element;
export default Layout;
