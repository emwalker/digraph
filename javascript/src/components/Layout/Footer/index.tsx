import React from 'react'
import { Link } from 'found'

import styles from './styles.module.css'

const Footer = () => (
  <footer className={styles.footer}>
    <div className="container-lg clearfix my-3 px-3 px-md-6 px-lg-3 my-6 pt-2 border-top">
      <p className="mb-2">
        <Link
          className="gray-link"
          to="/terms-of-use"
          test-id="terms-of-use"
        >
          Terms
        </Link>
        ・
        Software available under the MIT
        {' '}
        <a
          className="gray-link"
          href="https://github.com/emwalker/digraph/blob/master/LICENSE.md"
        >
          license
        </a>
        . © Eric Walker.
      </p>
    </div>
  </footer>
)

export default Footer
