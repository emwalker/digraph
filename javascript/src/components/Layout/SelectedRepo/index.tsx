import React from 'react'
import { graphql, useFragment } from 'react-relay'

import { SelectedRepo_viewer$key } from '__generated__/SelectedRepo_viewer.graphql'

type Props = {
  viewer: SelectedRepo_viewer$key,
}

export default function SelectedRepo(props: Props) {
  const viewer = useFragment(
    graphql`
      fragment SelectedRepo_viewer on User {
        selectedRepository {
          displayColor
          fullName
          isPrivate
        }
      }
    `,
    props.viewer,
  )

  const repo = viewer.selectedRepository
  if (!repo || !repo.isPrivate) return null
  const backgroundColor = repo.displayColor as string

  return (
    <div className="SelectedRepo-banner" style={{ backgroundColor }}>
      <h2>{repo.fullName || 'Private repo'}</h2>
    </div>
  )
}
