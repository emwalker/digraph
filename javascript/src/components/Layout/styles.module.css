.layout {
  min-height: 100vh;
  overflow: hidden;
  display: block;
  position: relative;
  padding-bottom: 140px;
}
