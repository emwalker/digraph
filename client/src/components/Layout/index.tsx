import React, { ReactNode } from 'react'
import { graphql } from 'react-relay'
import { Match, Router } from 'found'

import { LayoutQuery$data as Response } from '__generated__/LayoutQuery.graphql'
import FlashMessages from '../FlashMessages'
import Header from './Header'
import Footer from './Footer'
import SelectedRepo from './SelectedRepo'

type AlertsType = Response['alerts']
type ViewType = Response['view']

type Props = {
  alerts: AlertsType,
  children?: ReactNode,
  router: Router,
  match: Match,
  view: ViewType,
}

export const query = graphql`
  query LayoutQuery(
    $viewerId: ID!,
    $repoIds: [ID!],
    $searchString: String,
  ) {
    alerts {
      id
      text
      type
    }

    view(
      repoIds: $repoIds,
      searchString: $searchString,
      viewerId: $viewerId,
    ) {
      viewer {
        ...DesktopNav_viewer
        ...SelectedRepo_viewer
      }

      ...SearchBox_view
    }
  }
`

export default function Layout({ alerts, children, view, match, router }: Props) {
  return (
    <div className="layoutComponent">
      <Header
        location={match.location}
        router={router}
        view={view}
        viewer={view.viewer}
      />
      <div className="clearfix">
        <SelectedRepo viewer={view.viewer} />
        <FlashMessages initialAlertMessages={alerts} />
        { children }
      </div>
      <Footer />
    </div>
  )
}
