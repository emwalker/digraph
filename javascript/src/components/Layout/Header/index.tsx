import React from 'react'
import { Router } from 'found'

import { LocationType } from 'components/types'
import { LayoutQueryResponse as Response } from '__generated__/LayoutQuery.graphql'
import DesktopNav from './DesktopNav'

type ViewType = Response['view']
type ViewerType = ViewType['viewer']

type Props = {
  location: LocationType,
  router: Router,
  view: ViewType,
  viewer: ViewerType,
}

export default ({ location, router, viewer, view }: Props) => (
  <div className="outerHeader clearfix d-flex">
    <DesktopNav
      location={location}
      router={router}
      view={view}
      viewer={viewer}
    />
  </div>
)
