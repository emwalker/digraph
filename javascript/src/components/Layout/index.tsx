import React, { ReactNode } from 'react'
import { graphql } from 'react-relay'
import { Match, Router } from 'found'
import { MediaQueryProvider } from 'react-responsive-hoc'

import { LayoutQueryResponse } from '__generated__/LayoutQuery.graphql'
import FlashMessages from '../FlashMessages'
import Header from './Header'
import Footer from './Footer'
import styles from './styles.module.css'

type AlertsType = LayoutQueryResponse['alerts']
type ViewType = LayoutQueryResponse['view']

type Props = {
  alerts: AlertsType,
  children?: ReactNode,
  router: Router,
  match: Match,
  view: ViewType,
}

export const query = graphql`
query LayoutQuery(
  $viewerId: ID!,
  $orgLogin: String!,
  $repoName: String,
  $repoIds: [ID!],
  $searchString: String,
) {
  alerts {
    id
    text
    type
  }

  view(
    viewerId: $viewerId,
    currentOrganizationLogin: $orgLogin,
    currentRepositoryName: $repoName,
    repositoryIds: $repoIds,
    searchString: $searchString,
  ) {
    viewer {
      ...DesktopNav_viewer
      ...MobileNav_viewer
    }

    ...SearchBox_view
  }
}`

const Layout = ({ alerts, children, view, match, router }: Props) => (
  <MediaQueryProvider width={1600} height={800}>
    <div className={styles.layout}>
      <div>
        <Header
          location={match.location}
          router={router}
          view={view}
          viewer={view.viewer}
        />
        <div className="clearfix">
          <FlashMessages initialAlerts={alerts} />
          { children }
        </div>
        <Footer />
      </div>
    </div>
  </MediaQueryProvider>
)

export default Layout
