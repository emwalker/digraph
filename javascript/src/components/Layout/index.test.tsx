import React from 'react'
import { shallow } from 'enzyme'
import Layout from './index'

describe('<Layout />', () => {
  const router = {
    push: jest.fn(),
  }

  const match = {
    location: {
      query: { q: '' },
      search: '',
    },
  }

  const view = {
    viewer: {},
  }

  const wrapper = shallow(
    // @ts-expect-error
    <Layout router={router} match={match} view={view}>
      <div>some view</div>
    </Layout>,
  )

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('does not include links for the moment', () => {
    expect(wrapper.find('.test-links-page')).toHaveLength(0)
  })
})
