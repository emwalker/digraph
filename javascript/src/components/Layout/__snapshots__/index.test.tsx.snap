// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders 1`] = `
<MediaQueryProvider
  height={800}
  width={1600}
>
  <div
    className="layout"
  >
    <div>
      <_default
        location={
          Object {
            "query": Object {
              "q": "",
            },
            "search": "",
          }
        }
        router={
          Object {
            "push": [MockFunction],
          }
        }
        view={
          Object {
            "viewer": Object {},
          }
        }
        viewer={Object {}}
      />
      <div
        className="clearfix"
      >
        <FlashMessages />
        <div>
          some view
        </div>
      </div>
      <Footer />
    </div>
  </div>
</MediaQueryProvider>
`;
